Sprint 4

AlbertKhaydarov opened the pull request:
Выполнены требований ТЗ 
добавлена функция  setupBorder для логики скрытия рамки

AlbertKhaydarov committed:
Созданы расширения для UIColors, добавлены основные Images, добавлены шрифты, добавлен Launchscreen экран

AlbertKhaydarov committed:
Update task sprint 4

AlbertKhaydarov committed:
Добавлена логика работы с рамкой для изображения

amardanov commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,157 @@
 import UIKit
 
+struct QuizQuestion {
```
Советую всегда выносить модели в отдельные файлы по правилу "1 модель = 1 файл". Так навигация в коде становится проще и понятнее

AlbertKhaydarov replied:
вынес в отдельные файлы в группу Models

amardanov commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,157 @@
+    
+    private var currentQuestionIndex = 0
+    private var correctAnswers = 0
+    let padding = UIEdgeInsets(top: 5, left: 20, bottom: 5, right: 20)
```
Не увидел использования этого свойства дальше в файле. Думаю его стоит удалить тогда

AlbertKhaydarov replied:
удалил

amardanov commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,157 @@
+    @IBOutlet private weak var counterLabel: UILabel!
+    
+    // Mock-данные
+    private let questions: [QuizQuestion] = [QuizQuestion(imageOfFilm: "The Godfather",
```
Более понятно и аккуратно выглядит массив с таким перечислением, когда перечисление идет с другой строки, а открывающая скобка остается на той же. Закрывающая же наоборот, находится на строчке ниже относительно последнего элемента. Вот так:
```
private let questions: [QuizQuestion] = [
        QuizQuestion(imageOfFilm: "The Godfather",
                     questionText: "Рейтинг этого фильма\n больше чем 6?",
                     correctAnswer: true),
        QuizQuestion(imageOfFilm: "The Dark Knight",
                     questionText: "Рейтинг этого фильма\n больше чем 6?",
                     correctAnswer: true),
        QuizQuestion(imageOfFilm: "Kill Bill",
                     questionText: "Рейтинг этого фильма\n больше чем 6?",
                     correctAnswer: true),
        QuizQuestion(imageOfFilm: "The Avengers",
                     questionText: "Рейтинг этого фильма\n больше чем 6?",
                     correctAnswer: true),
        QuizQuestion(imageOfFilm: "Deadpool",
                     questionText: "Рейтинг этого фильма\n больше чем 6?",
                     correctAnswer: true),
        QuizQuestion(imageOfFilm: "The Green Knight",
                     questionText: "Рейтинг этого фильма\n больше чем 6?",
                     correctAnswer: true),
        QuizQuestion(imageOfFilm: "Old",
                     questionText: "Рейтинг этого фильма\n больше чем 6?",
                     correctAnswer: false),
        QuizQuestion(imageOfFilm: "The Ice Age Adventures of Buck Wild",
                     questionText: "Рейтинг этого фильма\n больше чем 6?",
                     correctAnswer: false),
        QuizQuestion(imageOfFilm: "Tesla",
                     questionText: "Рейтинг этого фильма\n больше чем 6?",
                     correctAnswer: false),
        QuizQuestion(imageOfFilm: "Vivarium",
                     questionText: "Рейтинг этого фильма\n больше чем 6?",
                     correctAnswer: false)
    ]
```

AlbertKhaydarov replied:
Форматирование массива - поправил

amardanov commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,157 @@
+    private var correctAnswers = 0
+    let padding = UIEdgeInsets(top: 5, left: 20, bottom: 5, right: 20)
+    
     override func viewDidLoad() {
```
Не забывай про форматирование кода в проекте. Отступы можно поправить с помощью сочетаний клавиш `cmd+a`, `control+i`. Советую пройтись и это сделать в каждом файле

AlbertKhaydarov replied:
4. Отступы во всех файлах выровнены

amardanov commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,157 @@
+        counterLabel.text = step.questionNumber
+    }
+    
+    private func setupBorder(cornerRadius: CGFloat, borderWidth: CGFloat){
```
Куда-то пропал пробел между 2 скобками в конце строчки)

AlbertKhaydarov replied:
добавил

AlbertKhaydarov committed:
Спасибо за комментарии!

Комментарии, полученные через  GitHub
1. Модели Вынесены в Отдельные файлы - группа Models (первоначально сделал все в одном файле, т.к. в ТЗ требование - «Весь код написан в одном файле MovieQuizViewController.swift»
2.     let padding = UIEdgeInsets(top: 5, left: 20, bottom: 5, right: 20)  - удалено
3.  Форматирование массива - поправил
4. Отступы во всех файлах выровнены
5. «Куда-то пропал пробел между 2 скобками в конце строчки)» - добавил

Комментарии через раздел «Ревью» в Практикум
1. Нужно исправить верстку (пункт 3.1) шрифты, к сожалению, неправильно импортированы. При открытии Storyboard файла показывается ошибка Font “YSDisplay-Bold” does not exist on this system, substituting the system font instead. Saving the document will lose the original font reference., указывающая на это. Советую посмотреть в сторону правильного импорта шрифтов и вообще ресурсов в проект

Ошибка не воспроизводится. Для контроля заново импортировал шрифты (в т.ч. переустановил шрифт на мой Mac) и переназначил на объектах. В «Build Phases»  .ttf файлы находятся в «Copy Bundle Resources». Через print Family Fonts выводятся в консоль. Дополнительно, в инфоплсте добавил явно расширение файла .ttf. В симуляторе  применяются.

2. В консоль пишется ошибка Failed to set (keyPath) user defined inspected property on (UIView): [<UIView 0x158f28a50> setValue:forUndefinedKey:]: this class is not key value coding-compliant for the key keyPath.. - исправлено.
3. «Обращу внимание, что пользователь может случайно прожать два раза кнопку ответа или даже много и быстро понажимать на кнопки» - добавил логику отключения кнопок (дополнительно проверил через isHidden))  )
